app/build.gradle

```diff
@@ -31,6 +33,14 @@ dependencies {
     implementation"org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
     implementation "androidx.appcompat:appcompat:1.0.0-rc02"
     implementation "androidx.constraintlayout:constraintlayout:2.0.0-alpha2"
+    implementation "androidx.lifecycle:lifecycle-extensions:$lifecycle_version"
+    implementation "androidx.lifecycle:lifecycle-common-java8:$lifecycle_version"
+    implementation "androidx.lifecycle:lifecycle-livedata:$lifecycle_version"
+    implementation "androidx.lifecycle:lifecycle-reactivestreams-ktx:$lifecycle_version"
+    implementation 'com.google.dagger:dagger:2.17'
+    annotationProcessor 'com.google.dagger:dagger-compiler:2.17'
```

Comment:
So first of all, this should be a kapt dependency, as we are in Kotlin. Moreover I see that this line is unnecessary, because you already have the kapt dependency.
Other than that, it would be better to extract the version of dagger to an ext field as you did with the lifecycle version.
Also because all of the dagger dependencies should use the same version.